pingcap/tiflow
# engine: support new task mode for dm

### What problem does this PR solve?


Issue Number: close https://github.com/pingcap/tiflow/issues/9230

### What is changed and how it works?
support new task mode for dm to get prepare for cloud import speeding up:
* dump: contains only dump progress
* load&sync: contains both load and sync progress

### Check List 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
  test using engine to deploy new task mode task. it works and passes sync-diff check.

#### Questions 

##### Will it cause performance regression or break compatibility?

##### Do you need to update user documentation, design documentation or monitoring documentation?

### Release note 

```release-note
None
```
